docs/adr/0002-database-corruption.md

```diff
@@ -0,0 +1,132 @@
+
+### A: Check for corruption whenever we perform a database operation (Decided)
+* Good, because this gives us the opportunity to fix corruption errors sooner.
+* Bad, because the consumer code probably doesn't expect the database to be
+  deleted and recreated in the middle of a query.  However, this is just an
+  extreme case of normal database behavior -- for example any given row can be
+  deleted during a sync.
+
+### B: Assume all errors during migration are from corrupt databases
+* Bad, because creates the possibility of deleting valid user data.
+* Bad, because logic errors are most likely during migrations.
+
+### B: Compare error values against known corruption errors for all operations
+* Good, because should eliminate the possibility of deleting valid user data.
+* Bad, because we don't currently have a good list of these.
+* Bad, because it's tricky to try to repair a database that the consumer code
+  is currently using.
+
+### B: Compare error values against known corruption errors when opening the database (Decided)
+* Good, because should eliminate the possibility of deleting valid user data.
+* Good, because from what we've seen on sentry it seems like this is a very common case.
+* Good, because inside the open() method is a very convenient time to try to repair the DB.
+* Bad, because we don't currently have a good list of these.
+
+### C: Delete the database file and recreate the database (Decided)
+* Good, because it would allow user's with corrupted databases to use the
+  affected components again
+* Bad, because any misidentification will lead to data loss.
+
+### C: Move the database file and recreate the database
+
+This option would be similar to 1, but instead of deleting the file we would
+move it to a backup location.  When we started up, we could look for backup
+files and try to import lost data.
+
+* Good, because if we misidentify corrupt databases, then we have the
+  possibility of recovering the data
+* Good, because it allows a way for users to delete their data (in theory).
+  If the consumer code executed a `wipe()` on the database, we could also
+  delete any backup data.
+* Bad, because it's very difficult to write a recovery function that merged
+  deleted data with any new data.  This function would be fairly hard to test
+  and it would be easy to introduce a new logic error.
+* Bad, because it adds significant complexity to the migration code
+* Bad, because the user experience would be strange.  A user would open the
+  app, discover that their data was gone, then sometime later discover that the
+  data is back again.
+
+### C: Return a failure code
+
+* Good, because this option leaves no chance of of user data being deleted
+* Good, because it's the simplest to implement
+* Bad, because the component will not be usable if the database is corrupt
+* Bad, because the user's data is potentially exposed in the corrupted database
+  file and we don't provide any way for them to delete it.
```

Comment:
I think this last point was a hangover from the migration focus and can probably be deleted - in the context of this document, it is the status-quo.